test/unit/services/awsAdapter_spec.js

```diff
@@ -148,4 +152,146 @@ describe("The AwsAdapter class ", function () {
 			});
 		});
 	});
+
+	describe("terminating an instance", function () {
+
+		var describeInstancesStub;
+		var terminateInstancesStub;
+		var waitForStub;
+
+		describe("when the instance is running", function () {
+			var result;
+			var awsAdapter = new AwsAdapter(ec2);
+
+			before(function () {
+				describeInstancesStub = Sinon.stub(ec2, "describeInstancesAsync", function () {
+					var data = { Reservations : [ { Instances : [ { InstanceId : VALID_AWS_ID } ] } ] };
+					return Bluebird.resolve(data);
+				});
+
+				terminateInstancesStub = Sinon.stub(ec2, "terminateInstancesAsync", function () {
+					var data = { TerminatingInstances : [ { InstanceId : VALID_AWS_ID } ] };
+					return Bluebird.resolve(data);
+				});
+
+				waitForStub = Sinon.stub(ec2, "waitForAsync", function () {
+					var data = { Reservations : [ { Instances : [ {
+						InstanceId : VALID_AWS_ID,
+						state      : "terminated"
+					} ] } ] };
+					return Bluebird.resolve(data);
+				});
+
+				return awsAdapter.terminateInstances(VALID_ID)
+				.then(function (response) {
+					result = response;
+				});
+
+			});
+
+			after(function () {
+				describeInstancesStub.restore();
+				terminateInstancesStub.restore();
+				waitForStub.restore();
+			});
+
+			it("terminates the instance, sets the state on the AWS console to terminated", function () {
+				expect(describeInstancesStub.args[ 0 ][ 0 ].Filters[ 0 ].Name).to.equal("tag:ID");
+				expect(describeInstancesStub.args[ 0 ][ 0 ].Filters[ 0 ].Values[ 0 ]).to.equal(VALID_ID);
+
+				expect(terminateInstancesStub.args[ 0 ][ 0 ].InstanceIds[ 0 ]).to.equal(VALID_AWS_ID);
+
+				expect(waitForStub.args[ 0 ][ 0 ]).to.equal("instanceTerminated");
+				expect(waitForStub.args[ 0 ][ 1 ].Filters[ 0 ].Name).to.equal("tag:ID");
+				expect(waitForStub.args[ 0 ][ 1 ].Filters[ 0 ].Values[ 0 ]).to.equal(VALID_ID);
+
+				expect(result.Reservations[ 0 ].Instances[ 0 ].InstanceId).to.equal(VALID_AWS_ID);
+				expect(result.Reservations[ 0 ].Instances[ 0 ].state).to.equal("terminated");
+			});
+
+		});
+
+		describe("when the instance has already been terminated (doesn't exist)", function () {
```

Comment:
I want to make sure that the intention of this test case is clear. This message (and the error message below) seems to imply that this test case should cover the case where the instance does not exist. However, in that case I would expect the `describeInstance()` call to fail since there is no instance to describe. Perhaps we should have two different test cases for the different scenarios?